pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -97,9 +97,17 @@ func (i *instantSplitter) MapExpr(expr parser.Expr, stats *MapperStats) (mapped
 	case *parser.BinaryExpr:
 		return i.mapBinaryExpr(e, stats)
 	case *parser.Call:
+		if isSubquery(e) {
+			// Subqueries are currently not supported by splitting, so we stop the mapping here.
+			return e, true, nil
+		}
```

Comment:
Can you explain what is this checking?

I see this function checks whether the first arg of a call is a subquery, but I don't understand why are we checking that (why only the first arg? why don't we just dig recursively?)

Reply:
I inherited this logic from the query sharding. See here:
[mimir/pkg/frontend/querymiddleware/astmapper/sharding.go](https://github.com/grafana/mimir/blob/cc0ef427c9c2e516d844479028c2c15a726be740/pkg/frontend/querymiddleware/astmapper/sharding.go#L95)

In PromQL functions I don't think we can have functions where a subquery in a position different than 0. Can we?

Reply:
`histogram_quantile` has the vector as the second argument

Reply:
Oh, but it has to be a range aggregation call. Well, still I think it's worth fixing the function to be future-proof instead of taking those assumptions. But I see the point now. Thanks.

Reply:
Actually `quantile_over_time(scalar, range-vector)` takes the `range-vector` as 2nd parameter. I'm currently trying to write a test to break the current logic and then I will fix it.

Reply:
Fixed here:
https://github.com/grafana/mimir/pull/2575/commits/1e88552d09a80e89cb2e6d714641e4e6719ab8f7